### PR TITLE
Aggressive Cows in Java

### PR DESCRIPTION
**Pull request by ritvij14**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?
Yes
(Write your answer here.)

### Description
This is the code to solve the famous SPOJ problem of Aggressive cows, which has been done with the help of Binary Search.
(Write your answer here.)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
Issue #875 
(Write your answer here.)

### CODE REVIEW

**Comment by ritvij14** [comment]
@siddharth25pandey @plazzy99 @iamrajiv @atarax665 @Kajol-Kumari  please review this.

**Comment by ritvij14** [comment]
Also, @siddharth25pandey @Kajol-Kumari  a humble request, please assign **Difficulty level** and  **CH20 label** to the issue for this PR, #875

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@ritvij14 move this program to `cp` section. Also, provide the problem link in comments.

**Comment by ritvij14** [comment]
@iamrajiv ok I will move it into the `cp` section. The link that you are talking about, is the link to the actual problem statement or the issue? The issue is #875 which needs CH20 and difficulty level tags.
I will add the link to the actual problem statement in the comments after Space-Time complexities.

**Comment by iamrajiv** [comment]
@ritvij14 I am talking of actual problem link i.e. https://www.spoj.com/problems/AGGRCOW/

**Comment by ritvij14** [comment]
@iamrajiv done.

**Comment by iamrajiv** [comment]
@ritvij14 move the program to `cp` section.

**Comment by ritvij14** [comment]
@iamrajiv sorry missed that. It's done now.

File: Java/other/AggressiveCows.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        return false;
+    }
+
+    private static int binSearch(int cows, int arr[]) {
+        int left = 0;
+        int right = arr[arr.length - 1] - arr[0];
+
+        // using binary search technique to find distances
+        while (left <= right) {
+            int mid = (left + right)/2;
```
Comment (iamrajiv) [inline Java/other/AggressiveCows.java]: @ritvij14 use `mid = left + ((right - left) / 2)` this will avoid overflow conditon.

**Review by iamrajiv — CHANGES_REQUESTED** [review]

File: Java/other/AggressiveCows.java (hunk 1 above)
Comment (ritvij14) [reply Java/other/AggressiveCows.java]: @iamrajiv done.

**Review by iamrajiv — APPROVED** [review]
@ritvij14 LGTM!

**Review by atarax665 — APPROVED** [review]
Looks good to me.

**Review by siddharth25pandey — APPROVED** [review]
@ritvij14 Looks good to me
Able to merge
Thanks!

**Comment by ritvij14** [comment]
@siddharth25pandey thank you for merging, could you also please add CH20 label to this PR or the issue #875?

**This issue was assigned by someone who is not a mentor for CH20 so he hasn't put the tag there, but all participants have taken part in it for CH20 only, also it has been made by a CH20 participant.**

**Pull request merged**